### PR TITLE
[Merged by Bors] - Validate topic name

### LINKED ISSUE #1464: [Bug]: Error when creating a topic with an underscore and/or an uppercase in the name
When I try to create a topic with an uppercase letter in the name I got this error:

```bash
Error: 
   0: Consumer Error
   1: Aug 19 15:26:09.586 DEBUG fluvio_socket::multiplexing: multiplexer terminated
Fluvio client error
   2: Fluvio socket error
   3: time out in serial: 1001 request: 4Aug 19 15:26:09.586 DEBUG fluvio::sync::controller: Topic received request
```

I see this in the SC logs
```
Aug 19 20:22:30.579 ERROR K8StateDispatcher{spec="Topic" namespace="acct-76ddb9e8-257e-4083-ae3b-8259065655e3"}:process_ws_action: fluvio_stream_dispatcher::dispatcher::k8_dispatcher: error: Topic, update spec client error: 422 Unprocessable Entity
```

K8 related documentation https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

### PR DESCRIPTION
**Pull request by morenol**
Fixes #1464

### CODE REVIEW

File: crates/fluvio-cli/src/topic/create.rs [hunk 1]
```diff
@@ -121,10 +121,15 @@ impl CreateTopicOpt {
             })
         };
 
-        let is_valid = hostname_validator::is_valid(&self.topic);
+        let is_valid = self
+            .topic
```
Comment (morenol) [inline crates/fluvio-cli/src/topic/create.rs:125]: Not sure where I should put a function with this validation logic, is it ok in fluvio-sc-schema? @sehz

File: crates/fluvio-cli/src/topic/create.rs [hunk 2]
```diff
@@ -121,10 +121,15 @@ impl CreateTopicOpt {
...
+            .chars()
+            .all(|ch| ch.is_ascii_lowercase() || ch.is_ascii_digit() || ch == '-')
+            && !self.topic.ends_with('-')
+            && !self.topic.starts_with('-');
         if !is_valid {
             return Err(CliError::InvalidArg(
-                "Topic name must only contain alphanumeric characters, '-', or '.'".to_string(),
+                "Topic name must only contain lowercase alphanumeric characters or '-', start with an alphanumeric character and end with an alphanumeric character.".to_string(),
```
Comment (sehz) [inline crates/fluvio-cli/src/topic/create.rs:132]: let's move this message to Error

File: crates/fluvio-sc/src/services/public_api/topic/create.rs [hunk 3]
```diff
@@ -75,6 +75,21 @@ pub async fn handle_create_topics_request<AC: AuthContext>(
 async fn validate_topic_request(name: &str, topic_spec: &TopicSpec, metadata: &Context) -> Status {
     debug!("validating topic: {}", name);
 
+    let valid_name = name
+        .chars()
+        .all(|ch| ch.is_ascii_lowercase() || ch.is_ascii_digit() || ch == '-')
+        && !name.ends_with('-')
+        && !name.starts_with('-');
```
Comment (sehz) [inline crates/fluvio-sc/src/services/public_api/topic/create.rs:82]: can you add unit test for this?

File: crates/fluvio-cli/src/topic/create.rs (hunk 2 above)
Comment (sehz) [reply crates/fluvio-cli/src/topic/create.rs:132]: this should move to client error

**Review by sehz — CHANGES_REQUESTED** [review]

File: crates/fluvio-cli/src/topic/create.rs (hunk 2 above)
Comment (morenol) [reply crates/fluvio-cli/src/topic/create.rs:128]: Not sure what you mean, in the client side this is wrapped behind the ApiError type with the status code and message https://github.com/infinyon/fluvio/blob/122fb9fd01a7dc6f89bb3c15285b9ca86c0d5add/crates/fluvio-sc-schema/src/response.rs#L46

Comment (sehz) [reply crates/fluvio-cli/src/topic/create.rs:128]: Error Message should be coming from ErrorCode (canonical).  ErrrorCode->ApiError->AdminError (Fluvio crate).

In the Fluvio Error, we have:
```
 #[error("Fluvio controlplane error: {0}")]
    AdminApi(#[from] ApiError),
```
We should just forward error message

Comment (morenol) [reply crates/fluvio-cli/src/topic/create.rs:128]: Got it, in this particular case, this error is from CLI parsing arguments, not sure if that applies to this, but I will change this on the SC handling side to include only the error code

**Comment by morenol** [comment]
bors ping

File: crates/fluvio-dataplane-protocol/src/error_code.rs [hunk 4]
```diff
@@ -58,6 +58,8 @@ pub enum ErrorCode {
     TopicInvalidConfiguration,
```
Comment (sehz) [inline crates/fluvio-dataplane-protocol/src/error_code.rs:58]: bump up version

File: crates/fluvio-sc-schema/src/topic/mod.rs [hunk 5]
```diff
@@ -1,5 +1,44 @@
 pub use fluvio_controlplane_metadata::topic::*;
```
Comment (sehz) [inline crates/fluvio-sc-schema/src/topic/mod.rs:1]: bump up crate version

**Review by sehz — CHANGES_REQUESTED** [review]

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**